app/models/forecast_snapshot.rb

```diff
@@ -4,12 +4,12 @@ class ForecastSnapshot < ActiveRecord::Base
   before_destroy :delete_files_from_disk
 
   # Get series name from series mnemonic
-  def retrieve_name(prefix)
-    m = Measurement.find_by(prefix: prefix.chomp('NS'))
-    if m.nil?
+  def retrieve_name(name)
+    s = Series.find_by(name: name)
+    if s.nil?
```

Comment:
I think in this case we should fall back to the measurement data_portal_name if it is not nil.